src/Microsoft.Diagnostics.Monitoring.EventPipe/EventPipeStreamProvider.cs

```diff
@@ -28,7 +30,43 @@ public async Task<Stream> ProcessEvents(DiagnosticsClient client, TimeSpan durat
             EventPipeSession session = null;
             try
             {
-                session = await client.StartEventPipeSessionAsync(_sourceConfig.GetProviders(), _sourceConfig.RequestRundown, _sourceConfig.BufferSizeInMB, cancellationToken).ConfigureAwait(false);
+                IEnumerable<EventPipeProvider> providers = _sourceConfig.GetProviders();
+                int bufferSizeInMB = _sourceConfig.BufferSizeInMB;
+                long rundownKeyword = _sourceConfig.RundownKeyword;
+                RetryStrategy retryStrategy = _sourceConfig.RetryStrategy;
+                bool retry = true;
+                while (retry)
+                {
+                    retry = false;
+                    try
+                    {
+                        EventPipeSessionConfiguration config = new(providers, bufferSizeInMB, (rundownKeyword != 0), true, rundownKeyword);
+                        session = await client.StartEventPipeSessionAsync(config, cancellationToken).ConfigureAwait(false);
+                    }
+                    catch (UnsupportedCommandException e)
+                    {
+                        if (retryStrategy == RetryStrategy.DropKeywordKeepRundown)
```

Comment:
If I'm reading this correct, both of these if blocks do exactly the same thing, so I don't understand what behavioral difference DropKeywordKeepRundown vs DropKeywordDropRundown is supposed to have.

Reply:
Good catch, that was a typo, the intent is in the `DropKeywordKeepRundown` branch to set the `rundownKeyword` to `EventPipeSession.DefaultRundownKeyword` instead of 0.